core/HmmModel.cc

```diff
@@ -118,7 +152,12 @@ void HmmModel2::BuildThetaGammaFst(vector<int64_t> &x, VectorFst<FstUtils::LogAr
 	}
 
 	// compute arc weight
-	double arcWeight = nlogGamma[yIM1][yI] + nlogTheta[yI][x[i]];
+	double arcWeight = nlogGamma[yIM1][yI];
+        if(!learningInfo->neuralRepFilename.empty()) {
+            arcWeight += getGaussianPDF(yI, neural[i]);
```

Comment:
call a utility function instead of a member function here. think about the appropriate function signature.